app/initializers/session.js

```diff
@@ -46,13 +46,13 @@ export default {
     app.deferReadiness();
 
     Ember.$.get(sessionUri).then(function(response) {
-      if (response.user) {
-        var user = Ember.merge(response.user, {
-          fullName: response.user.user_name,
-          profilePicture: response.user.profile_picture
-        });
+      var user = response.user;
+      
+      if (user) {
+        store.pushPayload('user', {user: user});
+        user = store.getById('user', user.id);
 
-        session.set('user', store.createRecord('user', user));
+        session.set('user', user);
       }
 
```

Comment:
@sluceno only for you to know about what happen here (and improve your Ember skills :) ); before we're creating an user when receiving the date from the session enpoint. The bad thing about that is that when you `user.save()` you will make a **POST** instead of and **UPDATE** because for EmberData the user is a new record.

Now we're just pushing the received data to the store see `pushPayload`, so, now the **save** method work as expected.